packages/layer-composer/src/generators/tile-cluster/tile-cluster.ts

```diff
@@ -0,0 +1,130 @@
+import { AnyLayer } from '@globalfishingwatch/mapbox-gl'
+import { Type, TileClusterGeneratorConfig, MergedGeneratorConfig } from '../types'
+import { isUrlAbsolute } from '../../utils'
+import { isConfigVisible } from '../utils'
+import { API_GATEWAY } from '../../layer-composer'
+import { Group } from '../..'
+
+export type GlobalTileClusterGeneratorConfig = Required<
+  MergedGeneratorConfig<TileClusterGeneratorConfig>
+>
+
+class TileClusterGenerator {
+  type = Type.TileCluster
+
+  _getStyleSources = (config: GlobalTileClusterGeneratorConfig) => {
+    if (!config.tilesUrl) {
+      throw new Error(`Tile Cluster layer should specify tilesUrl ${JSON.stringify(config)}`)
+    }
+    const tilesUrl = isUrlAbsolute(config.tilesUrl)
+      ? config.tilesUrl
+      : API_GATEWAY + config.tilesUrl
+    const url = new URL(tilesUrl.replace(/{{/g, '{').replace(/}}/g, '}'))
+    if (config.dataset) {
+      url.searchParams.set('datasets', config.dataset)
+    }
+    if (config.start && config.end) {
+      url.searchParams.set('dateRange', [config.start, config.end].join(','))
+    }
+    if (config.eventTypes) {
+      url.searchParams.set(
+        'types',
+        Array.isArray(config.eventTypes) ? config.eventTypes.join(',') : config.eventTypes
+      )
+    }
+
+    return [
+      {
+        id: config.id,
+        type: 'vector',
+        promoteId: 'event_id',
+        tiles: [decodeURI(url.toString())],
+      },
+    ]
+  }
+
+  _getStyleLayers = (config: GlobalTileClusterGeneratorConfig): AnyLayer[] => {
+    const layers = [
+      {
+        id: 'clusters',
+        type: 'circle',
+        source: config.id,
+        'source-layer': 'points',
+        filter: ['>', ['get', 'count'], 1],
+        paint: {
+          'circle-radius': [
```

Comment:
To discuss how to better calculate this sizes

Reply:
Always use square root when mapping a variable to the circle surface - which is what you really want

<img width="784" alt="Screenshot 2021-03-24 at 13 05 12" src="https://user-images.githubusercontent.com/1583415/112307727-95749180-8ca1-11eb-8d6b-8bce19d604fa.png">

https://blog.mapbox.com/announcing-expressions-in-gl-js-a72b55d0a6af

Then I suppose an exponential scale on top of that makes sense.